packages/fuselage-emitter/src/index.ts

```diff
@@ -0,0 +1,73 @@
+export type EventType = string | symbol;
+
+
+/**
+  * returned by an on/once method, can be used to turn `off` the subscription
+  * @public
+*/
+export type OffCallbackHandler = () => void;
+/**
+  * Handler callback
+  * @public
+*/
+export type Handler<T = any> = (event?: T) => void;
+
+type EventHandlerList = Array<Handler>;
+type EventHandlerMap = Map<EventType, EventHandlerList>;
+
+export interface Emitter {
+  on<T = any>(type: EventType, handler: Handler<T>): void;
+  once<T = any>(type: EventType, handler: Handler<T>): void;
+  off<T = any>(type: EventType, handler: Handler<T>): void;
+  emit<T = any>(type: EventType, event?: T): void;
+}
+
+
+const once = Symbol('once');
+
+/**
+ * Emitter Class
+ * @name emitter
+ * @returns {Emitter}
+ * @public
+ */
+export class Emitter implements Emitter {
+  private evts: EventHandlerMap = new Map();
+
+  private [once] = new WeakSet<Handler>();
+
+  has(key: EventType): boolean {
```

Comment:
Doesn't having a method to check if an event emitter has a listener inverts the event emitter logic? I mean, an event *must* be emitted regardless of having a listener or not. This method may encourage bad patterns, like writing logic that changes according to the structure of the emitter.